### PR TITLE
Server: Update references to server environment variables to reflect new name

### PR DESCRIPTION
**Pull request by Gabrielopesantos**
Updates the references to server environment variables, from prefix with "VAULT" to "BAO", in the documentation.
These variables were initially updated in the PR linked below.

Follow up to https://github.com/openbao/openbao/pull/157

## Target Release

Alpha

### CODE REVIEW

File: website/content/docs/platform/aws/lambda-extension.mdx
Comment (cipherboy) [inline website/content/docs/platform/aws/lambda-extension.mdx:1]: @JanMa I don't think we forked the AWS lambda extension, did we? I think that's what #47 talks about, which was inconclusive as we didn't have someone step forward and want to maintain it?

File: website/content/docs/platform/aws/lambda-extension.mdx
Comment (cipherboy) [inline website/content/docs/platform/aws/lambda-extension.mdx:209]: Technically per above, we didn't fork this and so it would still use the `VAULT_LOG_LEVEL`, but if Jan agrees, we might remove this instead. But I wouldn't hold up merge on this, my 2c.,

**Review by cipherboy — APPROVED** [review]
Looks great, thanks @Gabrielopesantos!

File: website/content/docs/platform/aws/lambda-extension.mdx
Comment (Gabrielopesantos) [reply website/content/docs/platform/aws/lambda-extension.mdx:209]: Oh, this is docs from an extension. I used a command to automatically replace everything at once and then went through each change to understand if they made change, this one might have escaped.

Comment (Gabrielopesantos) [reply website/content/docs/platform/aws/lambda-extension.mdx:209]: So if we don't fork the aws extension, which I agree we shouldn't (at least for now), we keep this documentation file as it was originally?

Comment (cipherboy) [reply website/content/docs/platform/aws/lambda-extension.mdx:209]: Either we keep it or we remove it entirely. I'd probably just delete it entirely for now, tbh. My 2c.

File: website/content/docs/platform/aws/lambda-extension.mdx
Comment (JanMa) [reply website/content/docs/platform/aws/lambda-extension.mdx:1]: So far we did not touch it. Therefore I also left the docs mostly unchanged for now. IMO if there's no-one willing to support it before GA, we should remove the docs.

File: website/content/docs/platform/aws/lambda-extension.mdx
Comment (Gabrielopesantos) [reply website/content/docs/platform/aws/lambda-extension.mdx:209]: Done :+1:

**Review by cipherboy — APPROVED** [review]
Looks good, thank you @Gabrielopesantos !

**Comment by naphelps** [comment]
@Gabrielopesantos There is a single commit that requires a --signoff for the DCO check to pass. https://github.com/openbao/openbao/pull/207/checks?check_run_id=23078806963

**Review by naphelps — APPROVED** [review]

**Pull request merged**
